src/app/merge-diff/merge-comparsion/merge-comparsion.component.ts

```diff
@@ -16,75 +16,143 @@ limitations under the License.
+  @ViewChild('targetContent') targetContent: ElementRef;
 
   @Input() source: Branchable;
   @Input() target: Branchable;
-  @Input() mergeItem : FullMergeItem;
+  @Input() mergeItem: FullMergeItem;
   @Input() isCommitting: boolean;
 
   @Output() cancelCommitEvent = new EventEmitter<MergeItem>();
   @Output() acceptCommitEvent = new EventEmitter<FullMergeItem>();
+  currentElement: string;
+  linkScroll = false;
 
-  constructor(private dialog: MatDialog) { }
+  sourceUsed: string;
+  committingContent: string;
+
+  constructor(private dialog: MatDialog) {}
+
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  ngOnChanges(changes: SimpleChanges): void {
+    if (this.isCommitting) {
+      this.sourceUsed = this.mergeItem.branchSelected;
+      switch (this.mergeItem.branchSelected) {
+        case MergeUsed.Mixed:
+          this.committingContent = this.mergeItem.mixedContent;
+          this.sourceUsed = 'Mixed';
+          break;
+        case MergeUsed.Target:
+          this.committingContent = this.mergeItem.targetValue;
+          this.sourceUsed = 'Target';
+          break;
+        default:
+          this.committingContent = this.mergeItem.sourceValue;
+          this.sourceUsed = 'Source';
+          break;
+      }
+    }
+  }
 
   ngOnInit(): void {
+
   }
 
-  cancelCommit()
-  {
+  cancelCommit() {
     this.cancelCommitEvent.emit(this.mergeItem);
   }
 
-  acceptCommit(branchUsed: MergeUsed)
-  {
-     this.mergeItem.branchSelected = branchUsed;
-     this.acceptCommitEvent.emit(this.mergeItem);
+  acceptCommit(branchUsed: MergeUsed) {
+    this.mergeItem.branchSelected = branchUsed;
+    this.acceptCommitEvent.emit(this.mergeItem);
   }
 
   openEditor() {
     // TODO: add in possible other editors, not just strings
     this.dialog
-      .open<ConflictEditorModalComponent, ConflictEditorModalData, ConflictEditorModalResult>(
+      .open<
         ConflictEditorModalComponent,
-        {
-          disableClose: true,
-          minWidth: '50%',
-          maxHeight: '98vh',
-          maxWidth: '98vw',
-          data: {
-            source: this.source,
-            target: this.target,
-            item: this.mergeItem
-          }
+        ConflictEditorModalData,
+        ConflictEditorModalResult
+      >(ConflictEditorModalComponent, {
+        disableClose: true,
+        minWidth: '50%',
+        maxHeight: '98vh',
+        maxWidth: '98vw',
+        data: {
+          source: this.source,
+          target: this.target,
+          item: this.mergeItem
         }
-      )
+      })
       .afterClosed()
-      .pipe(
-        filter(result => result.status === ModalDialogStatus.Ok)
-      )
-      .subscribe(result => {
-        alert(result.resolvedContent);
+      .pipe(filter((result) => result.status === ModalDialogStatus.Ok))
+      .subscribe((result: ConflictEditorModalResult) => {
+        if (result.status === ModalDialogStatus.Ok) {
```

Comment:
You shouldn't need this `if` block, the `filter()` operator in the pipe means only `ModalDialogStatus.Ok` results get returned.